### DISCUSSION TITLE
HackerLab Setup

### DISCUSSION THREAD

**Discussion by NepKitsuna** [WebGoat/WebGoat#1657] [Q&A] [answered]
I want to set up a virtual machine for an enclosed environment for this and other projects I would like to try. (Mainly to protect my machine from becoming vulnerable) Does it matter which OS I choose for the VM? I am trying to decide between Linux and Windows.

**Comment by zubcevic** [comment, marked as answer]
It really does not matter it will work on both. We always keep integration tests running on mac, linux and windows. Especially for the deserialization vulnerability. 
You can make a VM with native webgoat or through containers. There have been vagrant images and docker images to facilitate these things. But in general making such an image for a VM requires a bit more work to keep everything up to date. If you run it just for yourself, you can simply run it on localhost and prevent others from reaching your machine. If you want a lab like environment for mulitple users, then a VM based option would be an option if you want the benefit of the scoreboard and other multiuser features

**Reply by NepKitsuna** [reply]
Thank you! I didn't think it would matter too much but its good to know I can just run localhost. Thanks for the advice!
